### PR TITLE
feat(toc): seek anchor id from parent nodes when target is not available

### PR DESCRIPTION
**Pull request by chigix**
Signed-off-by: Richard Lea <chigix@zoho.com>

Thank you for creating a pull request to contribute to Hexo code! Before you open the request please review the following guidelines and tips to help it be more easily integrated:

- [ ] Add test cases for the changes.
- [x] Passed the CI test.

### CODE REVIEW

**Comment by YoshinoriN** [comment]
@chigix 
Thank you for your contribution.
I'm going to test this PR. Would you please tell me detail ?

**Comment by chigix** [comment]
@YoshinoriN Thanks for reviewing.

This change is mainly focusing on HTML like the case below:

```html
<div id="data-summarise" class="section level2">
<h2>Data Summarise</h2>
<p>...</p>
</div>
```

, where the id is not assigned directly on the heading tag. However, this id anchor is still work for navigating to the target heading. 

So something mocking the template like the case above could be testing case for this change.

File: lib/plugins/helper/toc.js [hunk 1]
```diff
@@ -21,8 +21,14 @@ function tocHelper(str, options = {}) {
   let lastLevel = 0;
 
   headings.each(function() {
+    function getId(ele) {
```
Comment (tcrowe) [inline lib/plugins/helper/toc.js]: Please bring this `getId` function above `headings.each(...)` scope.

Comment (chigix) [reply lib/plugins/helper/toc.js]: I see. I'll push tweaks later

Comment (tcrowe) [reply lib/plugins/helper/toc.js]: Thanks for your patience with us

**Comment by chigix** [comment]
@tcrowe I've pushed correction commits. May I have your review again?

**Comment by YoshinoriN** [comment]
@tcrowe 
Would you like some help? Should I test it locally?

**Comment by tcrowe** [comment]
Hey @YoshinoriN I tested it but I can't remember what the result was.

Do you have an easy way to get PR branches from contributors? Previously I was pulling from their repo but I think we can pull from PR number.

**Comment by YoshinoriN** [comment]
Hi @tcrowe 

How about this?

```sh
git fetch origin pull/3404/head:BRANCHNAME
```

**Comment by tcrowe** [comment]
@YoshinoriN Thanks! I'll try that method

**Comment by tcrowe** [comment]
Thank you @chigix for the enhancement and @YoshinoriN for the PR help 🙌

**Pull request merged**